tests/integration/app/test_routes.py

```diff
@@ -315,3 +321,65 @@ def test_json_builder_query(
                 assert len(json_res) == response
         except Exception:
             assert res.data.decode() == response
+
+
+class TestHarvestRecordRawAPI:
+    """Test the HarvestRecord API Raw endpoint."""
+
+    def test_xml_harvest_record_raw(
+        self,
+        interface,
+        organization_data,
+        source_data_waf_iso19115_2,
+        job_data_waf_iso19115_2,
+        client,
+    ):
+        """
+        Test the Raw endpoint for a harvest record with XML source.
+        The expectaion is that the XML data is returned as it appears in the
+        source_raw of the record.
+        """
+        interface.add_organization(organization_data)
+        interface.add_harvest_source(source_data_waf_iso19115_2)
+        harvest_job = interface.add_harvest_job(job_data_waf_iso19115_2)
+
+        harvest_source = HarvestSource(harvest_job.id)
+        harvest_source.prepare_external_data()
+        harvest_source.compare_sources()
+        harvest_source.write_compare_to_db()
+
+        iso2_name = "http://localhost:80/iso_2_waf/valid_iso2.xml"
+        iso2_test_record = harvest_source.external_records[iso2_name]
+        response = client.get(f"/harvest_record/{iso2_test_record.id}/raw")
+
+        assert response.status_code == 200
+        assert response.text == iso2_test_record.source_raw
+        assert response.json is None
```

Comment:
If the above suggestion is taken, then this needs to change to match

```suggestion
        assert response.json["raw"] == iso2_test_record.source_raw
```